schema/v1.0/ast/spans.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast // import "go.opentelemetry.io/otel/schema/v1.0/ast"
+
+import "go.opentelemetry.io/otel/schema/v1.0/types"
+
+// VersionOfSpans corresponds to a section representing a list of changes that happened
+// to spans schema in a particular version.
+type VersionOfSpans struct {
+	Changes []SpansChange
+}
+
+// VersionOfSpanEvents corresponds to a section representing a list of changes that happened
+// to span events schema in a particular version.
+type VersionOfSpanEvents struct {
+	Changes []SpanEventsChange
+}
+
+// SpansChange corresponds to a section representing spans change.
+type SpansChange struct {
+	RenameAttributes *RenameSpanAttributes `yaml:"rename_attributes"`
+}
+
+// SpanEventsChange corresponds to a section representing span events change.
+type SpanEventsChange struct {
+	RenameEvents     *RenameSpanEvents          `yaml:"rename_events"`
+	RenameAttributes *RenameSpanEventAttributes `yaml:"rename_attributes"`
+}
+
+type RenameSpanAttributes struct {
+	AttributeMap map[string]string `yaml:"attribute_map"`
+}
+
+type RenameSpanEvents struct {
+	EventNameMap map[string]string `yaml:"name_map"`
+}
+
+type RenameSpanEventAttributes struct {
+	ApplyToSpans  []types.SpanName  `yaml:"apply_to_spans"`
```

Comment:
I don't see this field in the OTEP.  Is there a newer definition for this schema that I should be looking at?

Reply:
Good catch. It is an OTEP bug. The text correctly says:
>In addition it is also possible to optionally specify spans and events that the transformation should apply to.

However, the block after that text is missing the `apply_to_spans` that the text alludes to. I wrote the text correctly but made a mistake when copy/pasting the yaml from my experimental codebase. Let me submit a patch to the OTEP. If the patch does not go through I will remove from here.

Reply:
https://github.com/open-telemetry/oteps/pull/180 is now merged.